graphql/collect_fields_cache_integration_test.go

```diff
@@ -0,0 +1,184 @@
+package graphql
```

Comment:
Added several tests to ensure caches are not hit unexpectedly.

Reply:
Extremely welcome! Thanks!